extensions/oplogPopulator/OplogPopulator.js

```diff
@@ -64,10 +63,61 @@ class OplogPopulator {
             const password = encodeURIComponent(authCredentials.password);
             cred = `${username}:${password}@`;
         }
-        let connectionUrl = `mongodb://${cred}${replicaSetHosts}/` +
+        this._mongoUrl = `mongodb://${cred}${replicaSetHosts}/` +
             `?w=${writeConcern}&readPreference=${readPreference}`;
-        if (replicaSet) {
-            connectionUrl += `&replicaSet=${replicaSet}`;
+        this._replicaSet = replicaSet;
+        this._database = database;
+
+        // Promisifying callback functions
+        this._MongoClientConnect = util.promisify(MongoClient.connect);
+    }
+
+    /**
+     * Connects to MongoDB using the MongoClientInterface
+     * and retreives the metastore collection
+     * @returns {Promise|undefined} undefined
+     * @throws {InternalError}
+     */
+    async _setupMongoClient() {
+        try {
+            const client = await this._MongoClientConnect(this._mongoUrl, {
+                 replicaSet: this._replicaSet,
+                 useNewUrlParser: true,
+            });
```

Comment:
can't we should pass `params.mongoConfig` ?
the param in backbeat config is the same format used by MongoClient, so this is usually enough I think

Reply:
We are not using arsenal's MongoClient here so params need to be adjusted
we are using the MongoClient provided by the mongodb library, this is done because we won't be using any of the arsenal implementation functions (not needed)